packages/connect/src/api/firmware/x509certificate.ts

```diff
@@ -286,6 +286,64 @@ const parseValidity = (asn1: Asn1) => {
     };
 };
 
+/*
+ * https://www.alvestrand.no/objectid/2.5.29.html
+ * Extensions ::= SEQUENCE {
+ *   extnID      OBJECT IDENTIFIER,
+ *   critical    BOOLEAN OPTIONAL,
+ *   extnValue   OCTET STRING
+ * }
+ */
+const parseExtensions = (data: Asn1) => {
+    const asn1 = derToAsn1(data.contents);
+    if (asn1.cls !== 0 || asn1.tag !== 16 || !asn1.structured) {
+        throw new Error("This can't be a Extension. Wrong data type.");
+    }
+
+    const readBoolean = (value?: Uint8Array) => value && value[0] === 0xff;
+    const readBit = (bytes: Uint8Array, byteIndex: number, bitIndex: number) =>
+        (bytes[byteIndex] >> (7 - bitIndex)) & 1;
+
+    const extensions: Array<Record<string, any>> = [];
+    derToAsn1List(asn1.contents).forEach(item => {
+        const [id, ...pieces] = derToAsn1List(item.contents);
+        if (id.cls !== 0 || id.tag !== 6 || id.structured) {
+            throw new Error('Bad extension. Does not begin with an OBJECT IDENTIFIER.');
+        }
+
+        const algorithm = derObjectIdentifierValue(id.contents);
+        const critical = pieces[0].tag === 1 ? readBoolean(pieces[0].contents) : false;
```

Comment:
Too be consistent, I would check that `pieces[0].cls === 0 and pieces[0].structured`.

Reply:
[5b93f05](https://github.com/trezor/trezor-suite/pull/13173/commits/5b93f05296d8b3e1dd9d7306bfee9ed64d8988ee)